### PR TITLE
Pytest config.py

### PR DESCRIPTION
**Pull request by AbbyTsai**
Pytest coverage for config.py. Makes progress on #1229 

```
----------- coverage: platform win32, python 3.7.4-final-0 -----------
Name                            Stmts   Miss  Cover   Missing
-------------------------------------------------------------
server\__init__.py                 38     12    68%   19-21, 37-45
server\config.py                   69      0   100%
server\csp.py                       1      0   100%
server\errors.py                   18      8    56%   11, 16-17, 22, 27-28, 33-34
server\feature_policy.py            1      0   100%
server\helpers.py                 126      0   100%
server\language.py                 31      1    97%   10
server\routes.py                   59     30    49%   12, 18, 24-26, 32, 38-42, 48, 58-61, 78-82, 90-92, 98, 103, 109-112, 118
server\tests\__init__.py            0      0   100%
server\tests\config_test.py        25      2    92%   17-18
server\tests\helpers_test.py      183      0   100%
server\tests\validate_test.py      52      0   100%
server\validate.py                 78     33    58%   25-53, 59-72, 81-82, 104, 108-109
-------------------------------------------------------------
TOTAL                             681     86    87%
Required test coverage of 85% reached. Total coverage: 87.37%.
```

Note after discussion with @bazzadp we removed Gravatar support as not used and doesn't even work when we tested it!

### CODE REVIEW

File: src/server/tests/config_test.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    SUPPORTED_YEARS, SUPPORTED_CHAPTERS, SUPPORTED_LANGUAGES
+
+
+test_year = ['2020', '2019', '2018']
+
+
+def test_get_config_year():
+    assert get_config(test_year[0]) is not None
+    assert get_config(test_year[1]) is not None
+    assert get_config(test_year[2]) is None
```
Comment (bazzadp) [inline src/server/tests/config_test.py:11]: I'm not seeing the value of this variable since we don't loop through it so think we should just simplify this with the following:

```suggestion
def test_get_config_year():
    assert get_config('2020') is not None
    assert get_config('2019') is not None
    assert get_config('2018') is None
```

File: src/server/tests/config_test.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+
+def test_static_avatar():
+    for contributor in get_config('2020'):
+        if contributor == 'michelleoconnor':
+            static_url = contributor['avatar_url']
+            assert static_url is not None and static_url.startswith('/static')
```
Comment (bazzadp) [inline src/server/tests/config_test.py:18]: Let's add a comment here to explain why we're hard coding a users:

```suggestion
def test_static_avatar():
    # For this test we use an existing contributor with no avatar as they should be assigned one on initial load
    for contributor in get_config('2020'):
        if contributor == 'michelleoconnor':
            static_url = contributor['avatar_url']
            assert static_url is not None and static_url.startswith('/static')
```

File: src/server/tests/config_test.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+
+def test_get_config_in_supported_year():
```
Comment (bazzadp) [inline src/server/tests/config_test.py:21]:
```suggestion
def test_get_config_for_all_supported_years():
```

File: src/server/tests/config_test.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    for year in SUPPORTED_YEARS:
+        assert get_config(year) is not None
+
+
+def test_get_chapters_in_supported_chapters():
```
Comment (bazzadp) [inline src/server/tests/config_test.py:26]:
```suggestion
def test_get_chapters_for_all_supported_chapters():
```

File: src/server/tests/config_test.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    for year in SUPPORTED_YEARS:
+        for chapter in get_chapters(get_config(year)):
+            assert chapter in SUPPORTED_CHAPTERS[year]
+
```
Comment (bazzadp) [inline src/server/tests/config_test.py:30]: Let's also add some specific tests in case `SUPPORTED_YEARS` or `SUPPORTED_CHAPTERS` are empty:

```suggestion


def test_get_chapters_good_chapter():
    assert 'javascript' in get_chapters(get_config('2019'))


def test_get_chapters_bad_chapter():
    assert 'random' not in get_chapters(get_config('2019'))

```

File: src/server/tests/config_test.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+def test_get_languages_in_supported_languages():
```
Comment (bazzadp) [inline src/server/tests/config_test.py:32]:
```suggestion
def test_get_languages_for_all_supported_languages():
```

File: src/server/tests/config_test.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    for year in SUPPORTED_YEARS:
+        for language in get_languages(get_config(year)):
+            assert language in SUPPORTED_LANGUAGES[year]
+
+
```
Comment (bazzadp) [inline src/server/tests/config_test.py:37]: Similar to above let's add some specific tests just in case `SUPPORTED_YEARS` or `SUPPORTED_LANGUAGES` is ever empty:

```suggestion


def test_get_languages_good_and_bad():
    languages = [str(language) for language in get_languages(get_config('2019'))]
    assert 'English' in languages
    assert 'random' not in languages
```

File: src/server/tests/config_test.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+def test_get_live():
+    for year in SUPPORTED_YEARS:
+        assert get_live(get_config(year)) is True
```
Comment (bazzadp) [inline src/server/tests/config_test.py:40]:
```suggestion
def test_get_live_for_all_supported_years():
    for year in SUPPORTED_YEARS:
        assert get_live(get_config(year)) is True


def test_get_live_for_known_years():
    assert get_live(get_config('2019')) is True
    assert get_live(get_config('2020')) is True


def test_get_live_for_bad_year_raises_error():
    with pytest.raises(TypeError):
        get_live(get_config('2018'))
```

**Review by bazzadp — CHANGES_REQUESTED** [review]
Very good first attempt @AbbyTsai !

I've suggested some improvements

**Review by bazzadp — APPROVED** [review]
